mlrun/projects/project.py

```diff
@@ -2135,14 +2137,40 @@ def get_run_status(
             notifiers=notifiers,
         )
 
+    # TODO: remove in 1.6.0
+    @deprecated(
+        version="1.4.0",
+        reason="'clear_context' will be removed in 1.6.0, this can cause unexpected issues",
+        category=FutureWarning,
+    )
     def clear_context(self):
         """delete all files and clear the context dir"""
-        if (
-            self.spec.context
-            and path.exists(self.spec.context)
-            and path.isdir(self.spec.context)
-        ):
-            shutil.rmtree(self.spec.context)
+        warnings.warn(
+            "This method deletes all files and clears the context directory or subdirectory (if defined)!"
+            "  Please keep in mind that this method can produce unexpected outcomes and is not recommended,"
+            " it will be deprecated in 1.6.0."
+        )
+        # clear only if the context path exists and not relative
+        if self.spec.context and os.path.isabs(self.spec.context):
+
+            # if a subpath is defined, will empty the subdir instead of the entire context
+            path_to_clear = (
+                path.join(self.spec.context, self.spec.subpath)
+                if self.spec.subpath
+                else self.spec.context
+            )
+            if path.exists(path_to_clear) and path.isdir(path_to_clear):
+                shutil.rmtree(path_to_clear)
+            else:
+                logger.warn(
+                    f"Attempt to clear {path} failed."
+                    f" Please ensure that your context or subdir is properly defined."
+                )
```

Comment:
the `if` checks the `path_to_clear` but you output `path` in the warning. 
I would suggest adding information about the path you resolved to provide the user with more information on the path we are trying to delete.

also eventually you know what the conditions that didn't met to delete it, so I would be more specific about that the path you looked for doesn't exists and isn't directory.